add tailscale guide (#1406)
jellyfin/jellyfin.org: docs/general/post-install/networking/2_tailscale.md

---
uid: network-tailscale
title: Tailscale
sidebar-position: 3
---

# Tailscale

> [Tailscale](https://tailscale.com) is a VPN based on Wireguard which offers easy configuration and high performance. It is great for users who don't want to open ports, or are behind CGNAT.

Reviewer: I am not a fan of the 2nd sentence 😅 
```suggestion
[Tailscale](https://tailscale.com) is a VPN based on Wireguard which offers easy configuration and high performance. It provides an effective alternative to using a reverse proxy, particularly in situations where opening ports is undesirable or not feasible, such as when the network is behind a carrier-grade NAT (CGNAT).
```

Author: I don't think your suggestion is appropriate, as you are implying reverse proxy cannot be used in conjunction with tailscale

Reviewer: I am implying the same usecase as you where - that a user is EITHER going to use tailscale to access their network, or have a reverse proxy internaly and an open port in the network. How about removing the reverse proxy part like so:

```suggestion
[Tailscale](https://tailscale.com) is a VPN based on Wireguard which offers easy configuration and high performance. It provides an effective alternative in situations where opening ports is undesirable or not feasible, such as when the network is behind a carrier-grade NAT (CGNAT).
```

## Prerequisites

A Tailscale account is required for Tailscale to work. An account can be created on the [Tailscale Website](https://tailscale.com)

## Using Tailscale Directly

Tailscale can be directly used to connect multiple devices together.

> Pros:

Author: What should even be in that table?

> Pros:

Author: I don't think it would be appropriate, since for each row the pro and the con would be unrelated things

- No changing router or network settings required.
- No Public IP or additional servers required.

Cons:

> 

Reviewer: imo a huge con is relying on a third party provider and tunneling your entire network through that!
Sure, they can be trusted, but you still are doing that, so to me that is a con!

Author: You are relying on a 3rd party provider the moment internet traffic leaves your home.

Reviewer: If tailscale servers are down, I will not be abled to access my self hosted stuff.
It is a seperate 3rd party provider that even requires login. I am AGAINST not pointing that out as a con, as it clearly is one for me!

Reviewer: This one actually is important to me.
How about adding:
```
- Requires a seperate third Party Account/ Provider
```

- Every client needs to have the Tailscale software installed
- Needs extra, complex configuration to block clients from connecting to each other.

Setup on server:

1. Install the Tailscale app on the server host and login. Instructions are available in [their install guide](https://tailscale.com/kb/1347/installation)

> 2. Enable Tailscale.

Reviewer: What do you mean by `Enable Tailscale`?
As someone who has never installed it, I am confused. 
You mean enabling the tailscale-service (e.g. via systemctl?)

3. Find and note your Tailscale IP of the server according to the [Tailscale Documentation](https://tailscale.com/kb/1033/ip-and-dns-addresses?tab=linux#finding-your-tailscale-ip-address). This IP will be used for other devices to connect. The IP should start with 100 (e.g. 100.12.34.56).

Setup on clients:

> 1. Install the Tailscale app on the client and login. Instructions are available in [their install guide](https://tailscale.com/kb/1347/installation)

Reviewer: ```suggestion
1. Install the Tailscale app on the client and login. Instructions are available in the [official install guide](https://tailscale.com/kb/1347/installation)
```
Its not clear who "they" are in the context. (imo)

Reviewer: Then do that.
Or "in their [official install guide]" maybe.
Official basically puts it down to either "Jellyfin" (us) or "Tailscale"(them) - If we then say "THEIR" its clear we mean tailscale (imo)

> 2. Enable Tailscale.

Reviewer: Same as above. What does "enabling tailscale" mean?

> 1. Install the Tailscale app on the client device. Instructions are available in [Tailscale's official install guide](https://tailscale.com/kb/1347/installation)

Author: why?

3. Type in the server IP from above into your browser or Jellyfin client of choice and connect. (e.g. 100.12.34.56:8096)

## Using Tailscale with a reverse proxy

> Tailscale can also be used to connect between a server and a remote reverse proxy. This is useful for getting around CGNAT or in situations where port forwarding is otherwise not possible.

Reviewer: Note that the user could also just set up a direct VPN connection in scenarios like this! 
This should also be noted here imo.

Author: Did you read the "using tailscale directly" section *that is right before this part*?

Reviewer: You don't understand what I am saying.
If you have a publicly reachable reverse proxy anyways, you can just install a wireguard server on that and VPN to THAT from your other end.

You dont need tailscale in that scenario at all. 
I think it should be noted when tailscale is not needed for something.

Author: Yes you don't need tailscale but tailscale makes life easier especially for beginners.

Reviewer: Imo still worth a note that you have that alternative!
Cant hurt to advertise it here! After all we are not doing advertising for tailscale BUT we want the users to have the best possible setup.

> 2. Login to the tailscale client and connect to the tailnet. On Windows and macOS that can be done through the icon in the system tray. On Linux that can be done with the comnand `sudo tailscale up`. On mobile devices that can be done through the tailscale app.

Reviewer: ```suggestion
2. Login to the Tailscale client and connect to the tailnet. On Windows and macOS that can be done through the icon in the system tray. On Linux that can be done with the comnand `sudo tailscale up`. On mobile devices that can be done through the tailscale app.
```

Setup on Jellyfin server:

1. Follow the `Setup on server` section in the [Using Tailscale Directly](#using-tailscale-directly) section to setup Tailscale on the Jellyfin server host.
2. Complete the setup on the reverse proxy server below.

> 3. Open the Dashboard of the Jellyfin server and go to Networking. Add the reverse proxy Tailscale IP to the `known proxies` setting. (e.g. 100.65.43.21)

Reviewer: This should refer to the reverse proxy guide too!
It explains in detail how to do that, and why it is required.
It also helps with setting up a reverse proxy specifically for Jellyfin.

Author: We already link to it a bit below

Reviewer: Where do we link to that guide? We only link to the caddy example!
I meant: https://jellyfin.org/docs/general/post-install/networking/reverse-proxy/#forwarded-for-headers

Author: and then we get users picking nginx proxy manager because ooh web interface and then fucking up the config *yet another time* showing up in the troubleshooting channels with a borked and half working half broken jellyfin server asking for help.

For real blake you have to consider *who* is even going to be reading this doc and *who* this doc is intended for.

Reviewer: What?
You need to put the Proxy Server as a known proxy!
The Guide I linked literally explains why you need to do that.

Author: if you have so many problems how about utilizing that button called "make suggestion"

> 3. Open the Dashboard of the Jellyfin server and go to Networking. Add the reverse proxy Tailscale IP to the `known proxies` setting. This is explained in detail in our [reverse proxy guide](./reverse-proxy/index.md#forwarded-for-headers).

Reviewer: ```suggestion
3. Open the Dashboard of the Jellyfin server and go to Networking. Add the reverse proxy Tailscale IP to the `known proxies` setting. This is explained in detail in our [reverse proxy guide](./8_reverse-proxy/index.md#forwarded-for-headers).
```

Sorry, my fault
